### PR TITLE
LF: add test for bigNumeric operations

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala [hunk 1]
```diff
@@ -234,23 +234,25 @@ object SValue {
     override def toString: String = s"SBigNumeric($value)"
   }
   object SBigNumeric {
-    // TODO https://github.com/digital-asset/daml/issues/8719
-    //   Decide what are the actual bound for BigDecimal
-    val MaxScale = 1 << 10
-    val MaxPrecision = MaxScale << 2
+    val MaxPrecision = 1 << 16
+    val MaxScale = MaxPrecision / 2
+    val MinScale = -MaxPrecision / 2 + 1
```
Comment (sofiafaro-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala:239]: Can we shift the `scale` range down by 1, so it's:

```
    val MaxScale = MaxPrecision / 2 - 1
    val MinScale = -MaxPrecision / 2
```

Instead? That way the `scale` range can be described exactly as the range of a signed 16-bit integer (`i16` in rust).  It's a range that most programmers are more familiar with.



image

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala [hunk 2]
```diff
@@ -366,7 +364,7 @@ class SBuiltinTest extends AnyFreeSpec with Matchers with TableDrivenPropertyChe
       }
     }
 
-    "Numeric binary operations compute proper results" in {
+    "Decimal binary operations compute proper results" in {
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala:367]: Why this change? Isn’t Decimal dead at the lf level?

**Review by cocreature — APPROVED** [review]
Great work thanks! I agree with @sofiafaro-da that it would be nice to stick to 16bit scale.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala:239]: We agree offline to keep 
```
    val MaxScale = MaxPrecision / 2
    val MinScale = -MaxPrecision / 2 + 1
```
as this preserves the nice symmetry:

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala [hunk 3]
```diff
@@ -234,23 +234,25 @@ object SValue {
     override def toString: String = s"SBigNumeric($value)"
   }
   object SBigNumeric {
-    // TODO https://github.com/digital-asset/daml/issues/8719
-    //   Decide what are the actual bound for BigDecimal
-    val MaxScale = 1 << 10
-    val MaxPrecision = MaxScale << 2
+    val MaxPrecision = 1 << 16
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala:237]: The bound `2^16` has been choose for 2 reasons: 
- it let us test of BigNumeric builtins with the largest possible values (`//daml-lf/interpreter:test_bignumeric` tooks around 3 mins)
- it is a nice power of 2

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala:367]: it is a glitch.

File: daml-lf/interpreter/BUILD.bazel [hunk 4]
```diff
@@ -42,7 +43,10 @@ da_scala_library(
 da_scala_test_suite(
     name = "tests",
     size = "small",
-    srcs = glob(["src/test/**/*.scala"]),
+    srcs = glob(
+        ["src/test/**/*.scala"],
+        exclude = ["src/test/**/SBuiltinBigNumericTest.scala"],
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/BUILD.bazel:48]: :+1:

**Comment by remyhaemmerle-da** [comment]
part of #8719
